### PR TITLE
Cleanup security/SDS interfaces

### PR DESCRIPTION
**Pull request by howardjohn**
Removes a lot of fields that are never set. Most of this is from gateway
SDS removal.

Also cleanup CSRSign interface. There was a ReqId field which doesn't
need to be part of the interface, as well as a misnamed field

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pkg/security/security.go [hunk 1]
```diff
@@ -200,8 +164,7 @@ type Options struct {
 // interface to get back a signed certificate. There is no guarantee that the SAN
 // in the request will be returned - server may replace it.
 type Client interface {
-	CSRSign(ctx context.Context, reqID string, csrPEM []byte, subjectID string,
-		certValidTTLInSec int64) ([]string /*PEM-encoded certificate chain*/, error)
+	CSRSign(ctx context.Context, csrPEM []byte, token string, certValidTTLInSec int64) ([]string, error)
```
Comment (dgn) [inline pkg/security/security.go:167]: nit: maybe move the comment that explains the cert format above the func def?

File: security/pkg/nodeagent/sds/server.go [hunk 2]
```diff
@@ -51,15 +49,11 @@ func NewServer(options *ca2.Options, workloadSecretCache ca2.SecretManager) (*Se
 	s := &Server{
```
Comment (dgn) [inline security/pkg/nodeagent/sds/server.go:49]: just a random thought: I was confused by the file path for a moment here - we don't ship a nodeagent anymore do we? I guess it's time we move the `sds` package a level up.

**Review by dgn — APPROVED** [review]

File: security/pkg/nodeagent/sds/server.go (hunk 2 above)
Comment (howardjohn) [reply security/pkg/nodeagent/sds/server.go:49]: Yep, a lot of cleanups needed here

**Comment by howardjohn** [comment]
@JimmyCYJ ptal

**Review by hzxuzhonghu — APPROVED** [review]

File: pkg/security/security.go (hunk 1 above)
Comment (dgn) [reply pkg/security/security.go:167]: nevermind the variable is even called `csrPEM`

**Review by JimmyCYJ — APPROVED** [review]

File: security/pkg/nodeagent/sds/server.go [hunk 3]
```diff
@@ -144,19 +133,10 @@ func (s *Server) initWorkloadSdsService(options *ca2.Options) error { //nolint:
...
 }
 
-func (s *Server) grpcServerOptions(options *ca2.Options) []grpc.ServerOption {
+func (s *Server) grpcServerOptions() []grpc.ServerOption {
 	grpcOptions := []grpc.ServerOption{
 		grpc.MaxConcurrentStreams(uint32(maxStreams)),
 	}
 
-	if options.CertFile != "" && options.KeyFile != "" {
-		creds, err := credentials.NewServerTLSFromFile(options.CertFile, options.KeyFile)
```
Comment (JimmyCYJ) [inline security/pkg/nodeagent/sds/server.go:153]: Do you know what is the context of adding this logic. What scenario was that?

Comment (howardjohn) [reply security/pkg/nodeagent/sds/server.go:153]: no clue, its from the original SDS pr

Comment (JimmyCYJ) [reply security/pkg/nodeagent/sds/server.go:153]: I see, this is probably safe to remove. Thanks.

**Pull request merged**